src/server/pkg/storage/chunk/reader.go

```diff
@@ -104,6 +124,77 @@ func (r *Reader) readChunk(chunk *Chunk) error {
 	return nil
 }
 
+// AtSplit registers a callback for when a chunk split point is encountered.
+// The callback is only executed at a split point found after reading WindowSize bytes.
+// The reason for this is to guarantee that the same split point will appear in the writer
+// the data is being written to.
+func (r *Reader) AtSplit(f func()) {
+	r.bytesBeforeSplit = 0
+	r.atSplit = f
+}
+
+func (r *Reader) executeAtSplitFunc() {
+	if r.atSplit != nil && r.bytesBeforeSplit > WindowSize {
+		r.atSplit()
+		r.atSplit = nil
+	}
+}
+
+// Copy is the basic data structure to represent a copy of data from
+// a reader to a writer.
+type Copy struct {
+	before, after *bytes.Buffer
```

Comment:
What do `before` and `after` represent here? A comment explaining them would be nice.

Reply:
I added a comment, in short they are the raw bytes before/after a set of full chunks. The concatention of the before, set of chunks, after forms the set of bytes represented by the copy.